tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -9097,17 +9097,15 @@ SpirvConstant *SpirvEmitter::getValueZero(QualType type) {
     QualType scalarType = {};
     if (isScalarType(type, &scalarType)) {
       if (scalarType->isSignedIntegerType()) {
-        return spvBuilder.getConstantInt(astContext.IntTy, llvm::APInt(32, 0));
+        return spvBuilder.getConstantInt(scalarType, llvm::APInt(32, 0));
       }
 
       if (scalarType->isUnsignedIntegerType()) {
-        return spvBuilder.getConstantInt(astContext.UnsignedIntTy,
-                                         llvm::APInt(32, 0));
+        return spvBuilder.getConstantInt(scalarType, llvm::APInt(32, 0));
```

Comment:
This case can be merged into the previous case now?

Reply:
Aha right. Nice catch :)

Reply:
Updated

Reply:
you can use
`scalarType->isIntegerType()` which checks for both signed and unsigned integers.
However, both `isUnsignedIntegerType()` and `isIntegetType()` return true for Booleans.

So I think it'd be best to use
`if(scalarType->isIntegerType() && !scalarType->isBooleanType())`

Reply:
Now it handles boolean case in a separate branch. PTAL.